xdcs-frontend/src/app/view/task-creation/task-creation.component.ts

```diff
@@ -0,0 +1,171 @@
+import { Component } from '@angular/core';
+import { TaskDefinitionDto } from '../../../api/models/task-definition-dto';
+import { TaskDefinitionsService } from '../../../api/services/task-definitions.service';
+import { ActivatedRoute } from '@angular/router';
+import { first, map } from 'rxjs/operators';
+import { NodesDto } from '../../../api/models/nodes-dto';
+import { NodesService } from '../../../api/services/nodes.service';
+import { KernelParamDto, NodeDto } from '../../../api/models';
+import { faChevronDown, faChevronUp, faPlus, faTrash } from '@fortawesome/free-solid-svg-icons';
+import { TreeNode } from 'angular-tree-component';
+import { TreeDirectory, TreeFileType } from '../../element/file-tree/file-tree.component';
+
+@Component({
+  selector: 'app-task-creation',
+  templateUrl: './task-creation.component.html',
+  styleUrls: ['./task-creation.component.less'],
+})
+export class TaskCreationComponent {
+  nodes: Array<NodeDto> = [];
+  taskDefinition: TaskDefinitionDto;
+  taskId: string;
+  files: Array<any> = [];
+  usedNodes: Array<NodeDto> = [];
+  private faChevronUp = faChevronUp;
+  private faChevronDown = faChevronDown;
+  private faTrash = faTrash;
+  private faPlus = faPlus;
+  private readonly loadHandler = path => this.loadFile(path);
+
+  constructor(private taskDefinitionsService: TaskDefinitionsService,
+              private route: ActivatedRoute,
+              private nodesService: NodesService) {
+
+  }
+
+  ngOnInit(): void {
+    this.route.params.subscribe(routeParams => {
+      this.loadTask(routeParams.id);
+    });
+    this.loadRoot();
+  }
+
+  private loadChildren(root: TreeNode): Promise<any[]> {
+    const path = this.nodeToPath(root);
+    return this.loadHandler(path).then(dir =>
+      dir.entries.map(entry => {
+        return {
+          name: entry.name,
+          model: {
+            name: entry.name,
+            type: entry.type,
+          },
+          hasChildren: entry.type === TreeFileType.DIRECTORY,
+        };
+      }));
+  }
+
+  private loadRoot() {
+    this.loadChildren(null).then(nodes => this.files = nodes);
+  }
+
+  private loadFile(path: string): Promise<TreeDirectory> {
+    function mapType(type: 'regular' | 'directory' | 'link'): TreeFileType {
```

Comment:
to jest copy-paste z task-definition

Reply:
w sumie strasznie dużo kodu to copy-paste

Reply:
done